### DISCUSSION TITLE
Examples on creating a LoRa mesh network

### DISCUSSION THREAD

**Discussion by dj-fiorex** [jgromes/RadioLib#675] [Q&A]
Hello everybody, first time using RadioLib and it seems really a beutiful project! I'm here to ask a simple question: i'm trying to create a LoRa Mesh network where every node is an Heltec Lora + Wifi V2 board, so every node can have also Internet connections; this way i can use LoRa for out of range node and retrasmit the messages from the node(s) with wifi to the LoRa only nodes. is there any example on how to do a mesh network with RadioLib? 
Thanks

**Comment by jgromes** [comment]
RadioLib is pretty much just a collection of drivers (with support for a couple basic protocols built in for convenience). However, what you're asking for is pretty high in the network stack, and is not implemented by RadioLib.

You can either try to create something yourself (if the network is just a couple of nodes then it should be pretty easy, e.g. adding a unique ID to each node and then implementing either path or logical addressing). However, there are projects built on top of RadioLib that implement mesh networking including routing, e.g. https://github.com/sudomesh/LoRaLayer2 which runs (or used to run, not sure about the status of the project) https://github.com/sudomesh/disaster-radio.

**Comment by dj-fiorex** [comment]
Thanks for the response! I already saw these projects, but i couldn't find some example on how to do "internet-connected-node" routing, in fact i need that all the nodes know which node has active internet connection and route msgs towards that node. But i think no one has this idea before

**Reply by jgromes** [reply]
I don't think it's an usual request - it seems like the logical conclusion of mesh networking. In my opinion, it seems similar to https://docs.openmqttgateway.com/

**Reply by dj-fiorex** [reply]
Yeah, I saw that project but I didn't find if and how open mqtt create a lora mesh node or not in their docs!
